### PR TITLE
Support arbitrary stack export values in python.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/pulumi/pulumi/issues/2448

### CODE REVIEW

File: sdk/python/lib/test/langhost/stack_output/__main__.py [hunk 1]
```diff
@@ -13,4 +13,23 @@
...
+pulumi.export("number", 1)
+pulumi.export("boolean", True)
+pulumi.export("list", [])
+pulumi.export("list_with_none", [None])
+pulumi.export("list_of_lists", [[], []])
+pulumi.export("set", set(["val"]))
+pulumi.export("dict", {"a": 1})
+pulumi.export("output", pulumi.Output.from_input(1))
+pulumi.export("class", TestClass())
+pulumi.export("recursive", recursive)
```
Comment (lukehoban) [inline sdk/python/lib/test/langhost/stack_output/__main__.py]: Small nit - this test doesn't appear to include any nested outputs inside other structures?

File: sdk/python/lib/pulumi/runtime/stack.py [hunk 2]
```diff
@@ -92,11 +94,92 @@ def __init__(self, func: Callable) -> None:
...
+
+    if isinstance(attr, Output):
+        return attr.apply(lambda v: massage(v, seen))
+
+    if hasattr(attr, "__dict__"):
+        # recurse on the dictionary itself.  It will be handled above.
+        return massage(attr.__dict__, seen)
+
+    # finally, recurse through iterables, converting into a list of massaged values.
+    return [massage(a, seen) for a in attr]
```
Comment (lukehoban) [inline sdk/python/lib/pulumi/runtime/stack.py]: Is it safe to do this at this point?  Are we sure that only iterable objects can get here?

Comment (CyrusNajmabadi) [reply sdk/python/lib/pulumi/runtime/stack.py]: yes, it should be safe due to the work here aligning with the is_primitive check.  i.e. we only get primitives if we have dicts/dict-likes/iterables.  we handle the first two above.

File: sdk/python/lib/test/langhost/stack_output/__main__.py (hunk 1 above)
Comment (CyrusNajmabadi) [reply sdk/python/lib/test/langhost/stack_output/__main__.py]: have added tests.

**Pull request merged**